pkg/webhook/policy.go

```diff
@@ -247,21 +257,37 @@ func (h *validationHandler) getDenyMessages(res []*rtypes.Result, req admission.
 				}
 				eventMsg := "Admission webhook \"validation.gatekeeper.sh\" denied request"
 				reason := "FailedAdmission"
-				if r.EnforcementAction == "dryrun" {
+				if enforcementAction == string(util.Dryrun) || enforcementAction == string(util.Warn) {
```

Comment:
"warn" and "dryrun" should have different messages.

With this new scheme, admins are likely to treat these two actions differently:

"dryrun" will be for changes that are under consideration but that they don't want to surface to the end user yet, pending analysis.

"warn" will be for things that are either too dangerous to enforce at the boundary or for pending changes that they want to warn users about.

Because of this, admins will want to be able to parse the results of each action separately.

Reply:
See below for warning message.

Reply:
I don't think this applies when we are talking about `eventMsg`.

the semantic logs and the events created by violations do not go through the admission response pipeline